metals/src/main/scala/scala/meta/internal/metals/FileDecoderProvider.scala

```diff
@@ -19,24 +22,36 @@ import scala.meta.internal.io.FileIO
 import scala.meta.internal.metals.MetalsEnrichments._
 import scala.meta.internal.metap.Main
 import scala.meta.internal.mtags.SemanticdbClasspath
+import scala.meta.internal.parsing.ClassFinder
 import scala.meta.io.AbsolutePath
 import scala.meta.metap.Format
 import scala.meta.metap.Settings
+import scala.meta.pc.PresentationCompiler
 
 import ch.epfl.scala.bsp4j.BuildTargetIdentifier
+import org.eclipse.{lsp4j => l}
 
+/* Response which is sent to the lsp client. Because of java serialization we cannot use
+ * sealed hierarchy to model union type of success and error.
+ * Moreover, we cannot use Option to indicate optional values, so instead every field is nullable.
+ * */
 final case class DecoderResponse(
-    requestedUri: String,
-    value: String,
-    error: String
+    @Nullable requestedUri: String,
```

Comment:
The `requestedUri` should always be there right? I don't believe this one needs to be `@Nullable`.

Reply:
Also, 🙏🏼 for aligning this ha. I was working on the `nvim-metals` implementation tonight of the semanticdb/javap stuff and realized that we have a slightly different structure is returned and I was actually going to shoot in a pr making this change for the tasty stuff to just use this.

Reply:
> The `requestedUri` should always be there right? I don't believe this one needs to be `@Nullable`.

You're right,`requestedUri`  shouldn't be `@Nullable`